From d7fbe90aefb7aeea1088555414d84c797b79fb27 Mon Sep 17 00:00:00 2001
From: Madhavan Ganesan <quic_madhgane@quicinc.com>
Date: Thu, 27 Jun 2024 15:13:33 +0530
Subject: [PATCH] PCI: qcom: enable async probe based on
 PCIE_ENABLE_ASYNC_PROBE

Add the Kconfig PCIE_ENABLE_ASYNC_PROBE to enable async probe type

Change-Id: I4776b9c36076881b8c6870e82d49428f6242b6de
Signed-off-by: Madhavan Ganesan <quic_madhgane@quicinc.com>
---
 drivers/pci/controller/dwc/Kconfig     | 5 +++++
 drivers/pci/controller/dwc/pcie-qcom.c | 2 ++
 2 files changed, 7 insertions(+)

diff --git a/drivers/pci/controller/dwc/Kconfig b/drivers/pci/controller/dwc/Kconfig
index 6fc2c7055cdb..07f4a49194a6 100644
--- a/drivers/pci/controller/dwc/Kconfig
+++ b/drivers/pci/controller/dwc/Kconfig
@@ -423,4 +423,9 @@ config PCIE_VISCONTI_HOST
 	  Say Y here if you want PCIe controller support on Toshiba Visconti SoC.
 	  This driver supports TMPV7708 SoC.
 
+config PCIE_ENABLE_ASYNC_PROBE
+	bool "Enable PCI Asynchronous probe type"
+	help
+	  Say Y to use Asynchronous probe, else Synchronous probe type.
+
 endmenu
diff --git a/drivers/pci/controller/dwc/pcie-qcom.c b/drivers/pci/controller/dwc/pcie-qcom.c
index e75754e6d7dd..908263ca096a 100644
--- a/drivers/pci/controller/dwc/pcie-qcom.c
+++ b/drivers/pci/controller/dwc/pcie-qcom.c
@@ -2108,7 +2108,9 @@ static struct platform_driver qcom_pcie_driver = {
 		.suppress_bind_attrs = true,
 		.of_match_table = qcom_pcie_match,
 		.pm = &qcom_pcie_pm_ops,
+#ifdef CONFIG_PCIE_ENABLE_ASYNC_PROBE
 		.probe_type = PROBE_PREFER_ASYNCHRONOUS,
+#endif
 	},
 };
 builtin_platform_driver(qcom_pcie_driver);
-- 
2.34.1

